File: solutions/035.search-insert-position/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
  let max = Math.max(...nums)

  if (nums.includes(target)) return nums.indexOf(target)
  else {
    if (target<max) {
      return nums.findIndex(e=> e>target)
    } else { return nums.length}
  }
}

File: solutions/035.search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        l = 0
        r = len(nums) - 1

        while l <= r:
            m = (l + r) // 2

            if nums[m] > target:
                # move r to m - 1
                r = m - 1
            elif nums[m] < target:
                l = m + 1
            else:
                return m

        return l
